src/neo/SmartContract/Native/Tokens/NeoToken.cs

```diff
@@ -50,45 +55,48 @@ protected override void OnBalanceChanging(ApplicationEngine engine, UInt160 acco
 
         private void DistributeGas(ApplicationEngine engine, UInt160 account, NeoAccountState state)
         {
-            BigInteger gas = CalculateBonus(state.Balance, state.BalanceHeight, engine.Snapshot.PersistingBlock.Index);
+            BigInteger gas = CalculateBonus(engine.Snapshot, state.Balance, state.BalanceHeight, engine.Snapshot.PersistingBlock.Index);
             state.BalanceHeight = engine.Snapshot.PersistingBlock.Index;
             GAS.Mint(engine, account, gas);
         }
 
-        private BigInteger CalculateBonus(BigInteger value, uint start, uint end)
+        private BigInteger CalculateBonus(StoreView snapshot, BigInteger value, uint start, uint end)
         {
             if (value.IsZero || start >= end) return BigInteger.Zero;
             if (value.Sign < 0) throw new ArgumentOutOfRangeException(nameof(value));
-            BigInteger amount = BigInteger.Zero;
-            uint ustart = start / Blockchain.DecrementInterval;
-            if (ustart < Blockchain.GenerationAmount.Length)
+
+            StorageKey keyLeft = CreateStorageKey(Prefix_GasPerBlock).Add(uint.MaxValue - end);
+            StorageKey keyRight = CreateStorageKey(Prefix_GasPerBlock).Add(uint.MaxValue);
+            var enumerator = snapshot.Storages.FindRange(keyLeft, keyRight).GetEnumerator();
```

Comment:
The integer in the key is little-endian. So you can't compare the keys. Maybe you can add a `AddBigEndian()` to replace `Add()`.

Reply:
Good catch, fixed.